Add Sway Section (#4)
FuelLabs/fuel-docs: src/sway-language.md

# Sway Language

> Sway is a smart contract programming language that prioritizes safety and speed and brings modern programming language theory to the blockchain. Check out the Sway docs [here](https://fuellabs.github.io/sway/v0.19.2/).

Reviewer: ```suggestion
Sway is a smart contract programming language that prioritizes safety and speed and brings modern programming language theory to the blockchain. Check out the Sway docs [here](https://fuellabs.github.io/sway/latest/).
```

> ## Rust + Solidity = Sway
> Sway prioritizes compile-time analysis and safety, similar to Rust’s borrow checker and safety-first semantics. Additionally, it has the syntax of Rust. From Solidity, Sway took the notion of a smart-contract-paradigm language with built-in top-level contract storage and blockchain mechanisms for ergonomic and safe contract programming.

Reviewer: Does the linter not complain about this? Surprising.

```suggestion
## Rust + Solidity = Sway

Sway prioritizes compile-time analysis and safety, similar to Rust’s borrow checker and safety-first semantics. Additionally, it has the syntax of Rust. From Solidity, Sway took the notion of a smart-contract-paradigm language with built-in top-level contract storage and blockchain mechanisms for ergonomic and safe contract programming.
```

In addition to robust documentation, tooling, clear error messages, and more – Sway brings the notion of static auditing to smart contracts. The compiler will catch things that one would typically hire an auditing firm to catch. This is particularly unique. In addition, Sway is highly performant and has extensible optimization passes and a modular backend for targeting different blockchain architectures.

> Check out Sway examples [here](https://github.com/FuelLabs/sway/tree/376010e7b8f84334ffacd0f99e64f185c971d213/examples).

Reviewer: While permalinks are usually preferred for referencing things that should be immutable, examples will be updated.

```suggestion
Check out Sway examples [here](https://github.com/FuelLabs/sway/tree/master/examples).
```
